File: src/unit-test-data/Data.Tests/Tests.DataTest/Data/Test.Data.RecordType.cs
using DeepEqual.Syntax;
using Xunit;

namespace PrimeFuncPack.UnitTest.Data.Tests;

partial class TestDataTest
{
    [Fact]
    public void PlusFifteenIdSomeStringNameRecord_ExpectIdIsFifteenAndNameIsSomeString()
    {
        var actual = TestData.PlusFifteenIdSomeStringNameRecord;

        var expected = new RecordType
        {
            Id = TestData.PlusFifteen,
            Name = TestData.SomeString
        };

        actual.ShouldDeepEqual(expected);
    }

    [Fact]
    public void PlusFifteenIdLowerSomeStringNameRecord_ExpectIdIsFifteenAndNameIsLowerSomeString()
    {
        var actual = TestData.PlusFifteenIdLowerSomeStringNameRecord;

        var expected = new RecordType
        {
            Id = TestData.PlusFifteen,
            Name = TestData.LowerSomeString
        };

        actual.ShouldDeepEqual(expected);
    }

    [Fact]
    public void MinusFifteenIdSomeStringNameRecord_ExpectIdIsMinusFifteenAndNameIsSomeString()
    {
        var actual = TestData.MinusFifteenIdSomeStringNameRecord;

        var expected = new RecordType
        {
            Id = TestData.MinusFifteen,
            Name = TestData.SomeString
        };

        actual.ShouldDeepEqual(expected);
    }

    [Fact]
    public void MinusFifteenIdNullNameRecord_ExpectIdIsMinusFifteenAndNameIsNull()
    {
        var actual = TestData.MinusFifteenIdNullNameRecord;

        var expected = new RecordType
        {
            Id = TestData.MinusFifteen
        };

        actual.ShouldDeepEqual(expected);
    }

    [Fact]
    public void ZeroIdNullNameRecord_ExpectIdIsZeroAndNameIsNull()
    {
        var actual = TestData.ZeroIdNullNameRecord;

        var expected = new RecordType();
        actual.ShouldDeepEqual(expected);
    }
}

File: src/unit-test-data/Data.FSharp.Tests/Tests.DataTest/Test.Data.RecordType.fs
namespace PrimeFuncPack.UnitTest.Data.Tests.TestData

open Xunit
open PrimeFuncPack.UnitTest
open TestHelper

module DataRecordTypeTest =

    [<Fact>]
    let ``TestData.PlusFifteenIdSomeStringNameRecord should be with the id is 15 and the name is the some string``() =
        let expected = RecordType(Id = TestData.PlusFifteen, Name = TestData.SomeString)
        TestData.PlusFifteenIdSomeStringNameRecord |> shouldBeDeepEqual expected

    [<Fact>]
    let ``TestData.PlusFifteenIdLowerSomeStringNameRecord should be with the id is 15 and the name is the some string``() =
        let expected = RecordType(Id = TestData.PlusFifteen, Name = TestData.LowerSomeString)
        TestData.PlusFifteenIdLowerSomeStringNameRecord |> shouldBeDeepEqual expected

    [<Fact>]
    let ``TestData.MinusFifteenIdSomeStringNameRecord should be with the id is -15 and the name is the some string``() =
        let expected = RecordType(Id = TestData.MinusFifteen, Name = TestData.SomeString)
        TestData.MinusFifteenIdSomeStringNameRecord |> shouldBeDeepEqual expected

    [<Fact>]
    let ``TestData.MinusFifteenIdNullNameRecord should be with the id is -15 and the name is null``() =
        let expected = RecordType(Id = TestData.MinusFifteen)
        TestData.MinusFifteenIdNullNameRecord |> shouldBeDeepEqual expected

    [<Fact>]
    let ``TestData.ZeroIdNullNameRecord should be with the id is 0 and the name is null``() =
        let expected = RecordType()
        TestData.ZeroIdNullNameRecord |> shouldBeDeepEqual expected